packages/graphql-hooks/src/useClientRequest.js

```diff
@@ -145,6 +144,18 @@ function useClientRequest(query, initialOpts = {}) {
         operationName: revisedOpts.operationName
       }
 
+      if (!isMounted.current) {
```

Comment:
Doesn't it feel weird to use `Promise.resolve`? instead of `Promise.reject` to return an error?

Reply:
@tiagoheliob it does feel a bit weird, but I was trying to conform to existing behavior as much as possible while providing a useful error message - the only other cases where this hook throws errors/rejects are input type checks, so I would find it a bit unexpected to get a throw here. If it is decided that it should reject instead, I'd be happy to update my PR, maybe @simoneb can chime in on this one?

Reply:
That's true, I agree!